packages/@aws-cdk/aws-rds/lib/private/parameter-group-family-mapping.ts

```diff
@@ -12,3 +14,28 @@ export interface ParameterGroupFamilyMapping {
    */
   readonly parameterGroupFamily: string
 }
+
+/**
+ * Get the latest parameter group family for this engine. Latest is determined using semver on the engine major version.
+ * When `engineVersion` is specified, return the parameter group family corresponding to that engine version.
+ * Return undefined if no parameter group family is defined for this engine or for the requested `engineVersion`.
+ */
+export function calculateParameterGroupFamily(
+  parameterGroupFamilies: ParameterGroupFamilyMapping[] | undefined,
+  engineVersion: string | undefined): string | undefined {
+  if (parameterGroupFamilies === undefined) {
+    return undefined;
+  }
+  if (engineVersion !== undefined) {
+    const family = parameterGroupFamilies.find(x => engineVersion.startsWith(x.engineMajorVersion));
```

Comment:
Is this guaranteed to only match one version? If not, is ordering guaranteed? Below, you sort the families by version, which implies this list isn't (guaranteed) sorted.

Reply:
I think it's guaranteed to match one version, provided the `engineMajorVersion` are unique, and not subsets of each other, but this is the case with the currently used major versions (and it makes sense, because it would be weird if `11` and `11.2` were both major versions ;p).